aspnetcore/aspnetcore-2.1.md

```diff
@@ -0,0 +1,202 @@
+
+For more information, see [Build Web APIs with ASP.NET Core](xref:web-api/index)
+
+## IHttpClientFactory
+
+The new `HttpClientFactory` type can be registered and used to configure and consume instances of `HttpClient` in applications. `HttpClient` already has the concept of delegating handlers that could be linked together for outgoing HTTP requests. The factory will:
+
+* Make registering of instances of `HttpClient` per named client more intuitive.
+* Implement a Polly handler that allows Polly policies to be used for Retry, CircuitBreakers, etc.
+
+For more information, see [Initiate HTTP Requests](xref:fundamentals/http-requests)
+
+## Kestrel on Sockets
+
+In ASP.NET Core 2.1, a new socket transport has been added to Kestrel. An application can be configured to use this new transport by adding the `.UseSockets()` extension to the `HostBuilder`
+
+```csharp
+public static IWebHost BuildWebHost(string[] args) =>
+    WebHost.CreateDefaultBuilder(args)
+    .UseSockets()
+    .UseStartup<Startup>()
+ ```
+
+## Generic host builder
+
+In ASP.NET Core 2.1, the Generic Host Builder (`HostBuilder`) has been introduced. This Builder can be used for applications that do not process HTTP requests (Messaging, background tasks, etc.).
+
+For more information, see [.NET Generic Host](xref:fundamentals/host/generic-host)
+
+## Updated SPA templates
+
+In ASP.NET Core 2.1, the Single Page Application templates for Angular, React, and React with Redux:
+
+* Are updated to use the standard project structures and build systems for each framework.
+
+The Angular template is based on Angular CLI, and the React templates are based on create-react-app.
+For more information, see [Use the Single Page Application templates with ASP.NET Core](xref:spa/index)
+
+## Microsoft.AspNetCore.App package
+
+ASP.NET Core 2.1 introduces a new `Microsoft.AspNetCore.App` meta-package. `Microsoft.AspNetCore.App` from `Microsoft.AspNetCore.All` in that it reduces the number of dependencies of packages not owned or supported by the ASP.NET or .NET teams. For more information, see [Microsoft.AspNetCore.App metapackage for ASP.NET Core 2.1](xref:fundamentals/metapackage-app).
+
+The `existing Microsoft.AspNetCore.All` meta-package will continue to be made available throughout the 2.x lifecycle. For more information, see [this announcement](https://github.com/aspnet/Announcements/issues/287).
+
+## Razor Pages search for Razor assets
+
+In 2.1, Razor Pages search for Razor assets (such as layouts and partials) in to following directorys in the listed order:
+
+1. Current Pages folder.
+1. */Pages/Shared/*
+1. */Views/Shared/*
+
+## Razor Pages in an area
+
+Razor Pages supports areas. To see an example of this, create a new Razor Pages web app with individual user accounts. A Razor Pages web app with individual user accounts includes */Areas/Identity/Pages*.
+
+## Potentially breaking changes
```

Comment:
Can we remove this section and instead point people to the migration doc in the following section?